modules/ROOT/pages/dataweave-language-introduction.adoc

```diff
@@ -399,6 +399,30 @@ In the Mule app XML, you can use the `${file::filename}` syntax to send a script
 </choice>
 ----
 
+[[dw_readers_writers]]
+== DataWeave Readers and Writers
+
+DataWeave can read input data as whole in-memory, in indexed fashion,
+or for some data formats, part-by-part by streaming the input. For large
+files, reading the entire file into memory can be slow and negatively impact
```

Comment:
The issue is more about running out of memory than being slow.

Reply:
fixed